### ISSUE TITLE
Allow sending Emails to non existing users

### ISSUE THREAD

**Issue by uhon** [FusionAuth/fusionauth-issues#743]
### Problem
FusionAuth API only allows sending emails for existing users.
If a service wants to offer a user invitation feature, allowing existing users to invite other people by specifying only their email address, then FusionAuth can't be used to send such an invitation email through its API. 
While the endpoint `POST /api/email/send/{emailTemplateId}` allows specifying additional CC and BCC addresses, it still requires a non-empty List of valid `userIds` to be provided in the request body.
See: https://fusionauth.io/docs/v1/tech/apis/emails#send-an-email

### Solution
The endpoint  `POST /api/email/send/{emailTemplateId}` should have an additional `to` parameter (optional) which allows one or more plain email-address to be specified. If this parameter is set, the `userIds` parameter can be empty (or missing). If both parameters are present both should be considered in the outgoing email.

### Alternatives/workarounds
Introduce a new endpoint like `POST /api/email/send-to-unknown/{emailTemplateId}` which allows to send emails to non-user email-addresses.

### Additional context
A short overview on workarounds suggested by @robotdan and @mooreds on slack:

```
@robotdan:
I believe you are correct, we have done that intentionally.
As a hack, you could create a dedicated user that is not designed to login. Each time you want to send an invite, you would call these APIs:
1. /api/user/update   - Change the name and email address to the target user to invite
2. /api/email/send     - Using the Id of the user you just updated
3. /api/user/update   - Revert the change from step 1 so that the user does not exist so they can sign up once invited.


@uhon:
This is unfortunate.
Your hack might work, but bares many dangers.
First of all, concurrency issues: all the steps have to run in a transaction, otherwise users might miss an email or even get one with the wrong requestData in it.
Also chaining these 3 api calls will take quite some time if many users are invited at once.
Also Users could be blocked from registration if step 3 fails. As a user who's just heard about a service (trough invitation email), I'd be very concerned if the system tells me, that an account with my email address already exists.
Since you're suggesting a hack, I think that intentional decision was questionable from beginning.

@robotdan:
The reason for the decision was that we didn’t intend to email users that did not exist in FusionAuth, at least not using the To: field. You can always just send the email outside of FusionAuth as well. Most SMTP providers provide APIs for this type of thing, or you can use MailChimp, or other transactional email providers.

@uhon:
Since I'll go the hacky path anyways. Wouldn't it be better to create a new user for each invitation and delete it afterwards?

@mooreds
Only thing you might need to be aware of is that it will affect global total user reporting. if you don't care, then that's probably not a big deal. You could also do a hybrid which would be: check if the dedicated user exists. if not, use it per steps 1,2,3 above. If so, create another dedicated user then do steps 1,2,3 with that user.
Basically building a pool of dedicated users, perhaps with a certain email domain (@example.com) so they'd be easy to search for.
But I would also strongly encourage you to file a feature request as Daniel suggests too. If others have your use case, it'd be nice for them to know the workaround and/or vote for implementation of a cleaner, less hacky solution.

@robotdan:
You could create a dedicated tenant so the users won’t collide if they go to sign up at a later time. And then configure that tenant to delete users that are not verified, so they will automatically be deleted.
```

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by harenadata** [comment]
Not sure I agree with the proposed solution of allowing any email in the To field, but want to chime in that the concept of an invitational feature is also very important to our platform and we're searching for implementation methods as well.  Would be nice to have native support for "invitation codes" which would be used to pre-fill some personal information and also could be passed back to the platform for post-registration process.

i.e. 
- Player A invites Player B to join the League.
- Player B receives email with a "Join Now" invitation link that directs them to screen to login or register on FusionAuth
- Player B completes either process, they are redirected to the league platform so the player can be automatically added to the league.

**Comment by robotdan** [comment]
Thanks for the feedback @harenadata this is helpful.

**Comment by ashkan-pm** [comment]
This would be an amazing addition to an already phenomenal system!

**Comment by robotdan** [comment]
I think this would be a reasonable addition - hopefully we can squeeze this in shortly.

**Comment by robotdan** [comment]
Internal notes:

1. Send API optionally takes `toAddresses`
2. Send API optionally takes `applicationId` which will result in the `application` object being available in the template. 
3. When `toAddresses` are provided, the `userIds` are optional.

**Comment by uhon** [comment]
Awesome. I'm glad seeing this implemented! You guys rock!
